src/osd/PG.cc

```diff
@@ -8558,7 +8558,7 @@ boost::statechart::result PG::RecoveryState::Active::react(const MNotifyRec& not
 		       << dendl;
     pg->proc_replica_info(
       notevt.from, notevt.notify.info, notevt.notify.epoch_sent);
-    if (pg->have_unfound()) {
+    if (pg->have_unfound() || pg->might_have_unfound.find(notevt.from) != std::end(pg->might_have_unfound)) {
```

Comment:
Wouldn't it be simpler to just do:

`pg->might_have_unfound.count(notevt.from)`

?

Reply:
yea indeed